chain/client/src/stateless_validation/chunk_endorsement/tracker_v2.rs

```diff
@@ -0,0 +1,132 @@
+use std::collections::HashMap;
+use std::num::NonZeroUsize;
+use std::sync::{Arc, Mutex};
+
+use lru::LruCache;
+use near_chain_primitives::Error;
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::sharding::ShardChunkHeader;
+use near_primitives::stateless_validation::chunk_endorsement::ChunkEndorsementV2;
+use near_primitives::stateless_validation::ChunkProductionKey;
+use near_primitives::types::AccountId;
+use near_primitives::version::ProtocolFeature;
+use near_store::Store;
+
+use crate::stateless_validation::validate::validate_chunk_endorsement;
+
+use super::ChunkEndorsementsState;
+
+// This is the number of unique chunks for which we would track the chunk endorsements.
+// Ideally, we should not be processing more than num_shards chunks at a time.
+const NUM_CHUNKS_IN_CHUNK_ENDORSEMENTS_CACHE: usize = 100;
+
+/// Module to track chunk endorsements received from chunk validators.
+pub struct ChunkEndorsementTracker {
+    epoch_manager: Arc<dyn EpochManagerAdapter>,
+    /// Used to find the chain HEAD when validating partial witnesses.
+    store: Store,
+    /// We store the validated chunk endorsements received from chunk validators.
+    /// Interior mutability is required to update the cache in process_chunk_endorsement & compute_chunk_endorsements.
+    chunk_endorsements: Mutex<LruCache<ChunkProductionKey, HashMap<AccountId, ChunkEndorsementV2>>>,
```

Comment:
nit: storing the whole `ChunkEndorsementV2` is not required here, we only need `ChunkEndorsementInner` and its signature, but that is not a big deal, feel free to keep it as it is for now

Reply:
Yeah, I'm thinking of keeping the full endorsement as it isn't too much of an overhead, and to store inner I'll have to expose inner as a pub field.